File: LeetCode/Single Number II/137.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) 
    {
        unordered_map<int,int> dict;
        for(const auto& elem : nums)
        {
            ++dict[elem];
            if(dict[elem] == 3) dict.erase(elem);
        }
        return (*dict.begin()).first;
    }
};

File: LeetCode/Single Number II/137.rs
use std::collections::HashMap;

impl Solution {
    pub fn single_number(nums: Vec<i32>) -> i32 {
        let mut dict = HashMap::<i32,i32>::new();
        nums.into_iter().for_each(|x| {
            *dict.entry(x).or_default() +=1;
            if(*dict.get(&x).unwrap() == 3) {dict.remove(&x);}
        });
        dict.into_iter().next().unwrap().0
        
    }
}
